Downgrade istio to 1.25

nrfox opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

    1. If this is your first time, please read our contributor guidelines: https://github.com/istio-ecosystem/sail-operator/blob/main/CONTRIBUTING.md
    2. Discuss your changes before you start working on them. You can open a new issue in the [Sail Operator GitHub repository](https://github.com/istio-ecosystem/sail-operator/issues) or start a discussion in the [Sail Operator Discussion](https://github.com/istio-ecosystem/sail-operator/discussions). By this way, you can get feedback from the community and ensure that your changes are aligned with the project goals.
    3. If the PR is unfinished, make is as a draft.
-->

#### What type of PR is this?
<!--
In order to minimize the time taken to categorize your PR, add a label accoutring to the PR type defined above.
Please, use the following labels, according to the PR type:
    * Enhancement / New Feature - enhancement
    * Bug Fix - bug
    * Refactor - cleanup/refactor
    * Optimization - enhancement
    * Test - test-e2e
    * Documentation Update - documentation
-->

- [ ] Enhancement / New Feature
- [ ] Bug Fix
- [ ] Refactor
- [ ] Optimization
- [ ] Test
- [ ] Documentation Update

#### What this PR does / why we need it:

Downgrades the `istio/istio` dep to match the 1.25 release branch. Result of running:
```
go get -u istio.io/istio@1.25.1
go mod tidy
```

#### Which issue(s) this PR fixes:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
Add related issue or PR if exists.
-->
Fixes #

Related Issue/PR #

#### Additional information:

FilipB commented:
I'm wondering if we need this when we still have `v1.25-alpha` version in https://github.com/istio-ecosystem/sail-operator/blob/main/pkg/istioversion/versions.yaml#L176

nrfox commented:
> I'm wondering if we need this when we still have v1.25-alpha version in https://github.com/istio-ecosystem/sail-operator/blob/main/pkg/istioversion/versions.yaml#L176

I removed the alpha version from `versions.yaml` and updated the bundle to `stable`.

istio-testing commented:
@nrfox: The following test **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:

Test name | Commit | Details | Required | Rerun command
--- | --- | --- | --- | ---
gencheck_sail-operator_release-1.25 | df4500aa7cfe62dd793c9611dc7fd36bcd6c5906 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio-ecosystem_sail-operator/761/gencheck_sail-operator_release-1.25/1909965918859956224) | true | `/test gencheck`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@nrfox: The following test **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:

Test name | Commit | Details | Required | Rerun command
--- | --- | --- | --- | ---
gencheck_sail-operator_release-1.25 | 3bccd80e9025f397a7c299e76fbc9368f8b39d9d | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio-ecosystem_sail-operator/761/gencheck_sail-operator_release-1.25/1909968247000666112) | true | `/test gencheck`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

nrfox commented:
/retest
